resources/lib/library.py

```diff
@@ -1279,12 +1345,46 @@ def addons( self ):
+                                path = "||BROWSE||" + item['addonid'].encode('utf-8')
+                                action = "RunAddOn(" + item['addonid'].encode('utf-8') + ")"
+                            elif contenttype == "executable":
+                                # Check if it's a program that can be run as an exectuble
+                                provides = self.hasPluginEntryPoint( item[ "path" ] )
+                                if provides is None:
+                                    continue
+                                
+                                for content in provides:
+                                    # For each content that it provides, add it to the add-ons for that type
+                                    if content == "video":
+                                        otherItem = self._create([path, item['name'] + "  >", "::SCRIPT::32010", {"icon": "DefaultAddon.png", "thumb": thumb} ])
+                                        otherItem.setProperty( "path", "||BROWSE||" + item['addonid'].encode('utf-8') )
+                                        otherItem.setProperty( "action", "RunAddOn(" + item['addonid'].encode('utf-8') + ")" )
+                                        videoItems[ item[ "name" ] ] = otherItem
+                                    elif content == "audio":
+                                        otherItem = self._create([path, item['name'] + "  >", "::SCRIPT::32011", {"icon": "DefaultAddon.png", "thumb": thumb} ])
+                                        otherItem.setProperty( "path", "||BROWSE||" + item['addonid'].encode('utf-8') )
+                                        otherItem.setProperty( "action", "RunAddOn(" + item['addonid'].encode('utf-8') + ")" )
+                                        audioItems[ item[ "name" ] ] = otherItem
+                                    elif content == "image":
+                                        otherItem = self._create([path, item['name'] + "  >", "::SCRIPT::32012", {"icon": "DefaultAddon.png", "thumb": thumb} ])
+                                        otherItem.setProperty( "path", "||BROWSE||" + item['addonid'].encode('utf-8') )
+                                        otherItem.setProperty( "action", "RunAddOn(" + item['addonid'].encode('utf-8') + ")" )
+                                        imageItems[ item[ "name" ] ] = otherItem
+                                    elif content == "executable":
+                                        otherItem = self._create([path, item['name'] + "  >", shortcutType, {"icon": "DefaultAddon.png", "thumb": thumb} ])
+                                        otherItem.setProperty( "path", "||BROWSE||" + item['addonid'].encode('utf-8') )
+                                        otherItem.setProperty( "action", "RunAddOn(" + item['addonid'].encode('utf-8') + ")" )
+                                        listitems[ item[ "name" ] + ".1" ] = otherItem
+
```

Comment:
this block probably could also get done with less and non-repeating code by using a list of tuples?